pyproject.toml

```diff
@@ -1,3 +1,27 @@
+# Specification of pyproject.toml:
+# https://packaging.python.org/en/latest/guides/writing-pyproject-toml/
+
+# Should pyproject.toml be added?
+# A pyproject.toml file is strongly recommended. The presence of a pyproject.toml 
+# file itself does not bring much. What is actually strongly recommended is the 
```

Comment:
It does bring a lot, actually. Many tools take the presence of `[build-system]` as an explicit opt-in for switching to the isolated PEP 517 mode (usually the ones that used to support invoking setuptools directly in the past through calling `setup.py`).

Moreover, if the `build-backend` is unspecified, the `setuptools.build_meta:__legacy__` is the standardized fallback. And it works differently than the modern/explicit `setuptools.build_meta`.

Reply:
then also the docs should be changed? because I was taking all info straight from the docs

Reply:
Do you have a specific link? Did you copy it as is? Did you change it taking the context into account?

Reply:
Ah. Found it in the PR description. It would be useful to add such links right in the diff, next to each change in the patch to eliminate the mental overhead of matching the PR description with the patch that is displayed on a different PR tab.

Reply:
Hey @sinoroc, you wrote that text IIRC. Could you take a look and consider clarifying the PyPUG document?

Reply:
:)

I'm not against including the links right in the file! 
(I thought I'd find opposition here due to overcrowding it with links)

Imagine how I feel when there are 7 different places to read about python packaging)

Reply:
It's not about including the links in the code comments. It's about annotating your pull request with context. Using inline discussion threads, like this one.

Reply:
So from the user point of view:

`[build-system]` table exists --(implies)--> `build-backend` is specified 
(because of the comment `# REQUIRED if [build-system] table is used`)

Then the user might think, "and what if I don't include the `[build-system]` table?"
And the importance of having a `[build-system]` table is correctly stated in the comment:
```py
# The presence of a pyproject.toml 
# file itself does not bring much. What is actually strongly recommended is the 
# [build-system] table in pyproject.toml
```

---

Do you agree that "the presence of a pyproject.toml file itself (i.e. without the `[build-system]` table) does not bring much"? We can remove only this sentence if it's incorrect

Reply:
I have been mentioned, so here is my take:

I did not understand what is the "issue" with the sentence. Seems fine to me. `pyproject.toml` in itself does not do much (or at least it should not do much, at some point in time its presence changed the behavior of things for other tools but I think it has been fixed for a wile now).

To be honest I do not think we need all these prose in there. I guess I would only put the links to the [guide](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/) (now it says "specification" but it is a link to a guide) and then the [specification](https://packaging.python.org/en/latest/specifications/pyproject-toml/).